x/erc20/keeper/ibc_callbacks_test.go

```diff
@@ -2,574 +2,482 @@ package keeper_test
-
-				// for testing purposes we can only fund is not allowed to receive funds
-				moduleAcc := suite.app.AccountKeeper.GetModuleAccount(suite.ctx, "erc20")
-				sender = moduleAcc.GetAddress()
-				err := testutil.FundModuleAccount(
-					suite.ctx,
-					suite.app.BankKeeper,
-					moduleAcc.GetName(),
-					sdk.NewCoins(
-						sdk.NewCoin(pair.Denom, math.NewInt(100)),
-					),
-				)
-				suite.Require().NoError(err)
-
-				ack = channeltypes.NewErrorAcknowledgement(errors.New(""))
-				data = transfertypes.NewFungibleTokenPacketData("", "", sender.String(), "", "")
-			},
-			expPass:  true,
-			expERC20: big.NewInt(0),
+			name:          "error - invalid sender",
+			transfer:      transfertypes.FungibleTokenPacketData{Sender: ""},
+			expectedError: errors.New("empty address string is not allowed"),
 		},
 		{
-			name: "no-op - positive ack",
-			malleate: func() {
-				// Register Token Pair for testing
-				pair = suite.setupRegisterCoin(metadataCoin)
-				suite.Require().NotNil(pair)
-
-				sender = sdk.AccAddress(senderPk.PubKey().Address())
-
-				// Fund receiver account with EVMOS, ERC20 coins and IBC vouchers
-				// We do this since we are interested in the conversion portion w/ OnRecvPacket
-				err := testutil.FundAccount(
-					suite.ctx,
-					suite.app.BankKeeper,
-					sender,
-					sdk.NewCoins(
-						sdk.NewCoin(pair.Denom, math.NewInt(100)),
-					),
-				)
-				suite.Require().NoError(err)
-
-				ack = channeltypes.NewResultAcknowledgement([]byte{1})
-			},
-			expERC20: big.NewInt(0),
-			expPass:  true,
+			name: "noError - pair not registered",
+			transfer: transfertypes.NewFungibleTokenPacketData(
+				"nocoin",
+				"100",
+				sender,
+				receiver,
+				"",
+			),
+			expectedError: nil,
 		},
 	}
 	for _, tc := range testCases {
-		suite.Run(fmt.Sprintf("Case %s", tc.name), func() {
-			suite.SetupTest() // reset
-
-			tc.malleate()
-
-			err := suite.app.Erc20Keeper.OnAcknowledgementPacket(
-				suite.ctx, channeltypes.Packet{}, data, ack,
+		suite.Run(tc.name, func() {
+			err := unitNetwork.App.Erc20Keeper.OnTimeoutPacket(
+				unitNetwork.GetContext(),
+				channeltypes.Packet{},
+				tc.transfer,
 			)
-			suite.Require().NoError(err)
 
-			if tc.expPass {
-				suite.Require().NoError(err)
+			if tc.expectedError != nil {
+				suite.Require().Equal(tc.expectedError.Error(), err.Error())
 			} else {
-				suite.Require().Error(err)
+				suite.Require().Equal(tc.expectedError, err)
 			}
 
-			// check balance is the same as expected
-			balance := suite.app.Erc20Keeper.BalanceOf(
-				suite.ctx, contracts.ERC20MinterBurnerDecimalsContract.ABI,
-				pair.GetERC20Contract(),
-				common.BytesToAddress(sender.Bytes()),
-			)
-			suite.Require().Equal(tc.expERC20.Int64(), balance.Int64())
 		})
 	}
 }
 
-func (suite *KeeperTestSuite) TestOnTimeoutPacket() {
-	senderAddr := "evmos1x2w87cvt5mqjncav4lxy8yfreynn273xn5335v"
+func (suite *Erc20KeeperTestSuite) OnAcknowledgementPacket() {
```

Comment:
nit: other tests usually begin with `Test...`
```suggestion
func (suite *Erc20KeeperTestSuite) TestOnAcknowledgementPacket() {
```

Reply:
done